### DISCUSSION TITLE
Migration to latest version of react js

### DISCUSSION THREAD

**Discussion by AbeykoonOshan** [react-grid-layout/react-grid-layout#1727] [General]
Does react-grid-layout compatible with React version 18?

**Comment by bswan0002** [comment]
There's at least one issue in React 18, see this Issue: https://github.com/react-grid-layout/react-grid-layout/issues/1745

I'll note that I have a project where my grid layout breaks when updating to React 18. However, this is likely due to some spaghetti code I wrote that is breaking with React 18 batched state updates rather than an issue with the grid layout itself.

It would be cool to see support for 18, though.

**Comment by LoicKairon** [comment]
I am using react-grid-layout in a typescript project and I am getting types error with the new version of React:
```
'ResponsiveGridLayout' cannot be used as a JSX component.
```

Apparently, due to a `@types/react` dependency as discussed [here](https://github.com/facebook/react/issues/24304).

Is there already a fix for this?

**Reply by hardikchopra242** [reply]
Have you found any fix for this?

**Reply by LoicKairon** [reply]
Yes, probably not the best fix though, but it has been working without issues since then:

```Typescript
const ResponsiveGridLayout = WidthProvider(Responsive) as unknown as (
  props: ResponsiveProps,
) => JSX.Element;
```

```
"react": "^18.2.0"
"react-grid-layout": "^1.3.4"
"@types/react-grid-layout": "^1.3.2"
```

**Reply by hardikchopra242** [reply]
Thank you so much @LoicKairon

**Comment by KeithHubert** [comment]
Just wanted to check if there were any updates regarding react 18 support.

We loved using the library in an older project, but I'd feel better about using it in a new one if it was confirmed that support for 18 was in the works. 

Thanks.
